_includes/v2.2/app/insecure/sqlalchemy-basic-sample.py

```diff
@@ -1,35 +1,108 @@
-from __future__ import print_function
+import random
+from math import floor
 from sqlalchemy import create_engine, Column, Integer
 from sqlalchemy.ext.declarative import declarative_base
 from sqlalchemy.orm import sessionmaker
+from cockroachdb.sqlalchemy import run_transaction
 
 Base = declarative_base()
 
+
 # The Account class corresponds to the "accounts" database table.
 class Account(Base):
     __tablename__ = 'accounts'
     id = Column(Integer, primary_key=True)
     balance = Column(Integer)
 
-# Create an engine to communicate with the database. The "cockroachdb://" prefix
-# for the engine URL indicates that we are connecting to CockroachDB.
-engine = create_engine('cockroachdb://maxroach@localhost:26257/bank',
-                       connect_args = {
-                           'sslmode' : 'disable'
-                       })
-Session = sessionmaker(bind=engine)
+
+# Create an engine to communicate with the database. The
+# "cockroachdb://" prefix for the engine URL indicates that we are
+# connecting to CockroachDB using the 'cockroachdb-python' dialect.
+# For more information, see
+# https://github.com/cockroachdb/cockroachdb-python.
+
+engine = create_engine(
+    'cockroachdb://maxroach@localhost:26257/bank',
+    connect_args={
+        'sslmode': 'disable',
+    },
+    echo=True                   # Log SQL queries to stdout
+)
 
 # Automatically create the "accounts" table based on the Account class.
 Base.metadata.create_all(engine)
 
-# Insert two rows into the "accounts" table.
-session = Session()
-session.add_all([
-    Account(id=1, balance=1000),
-    Account(id=2, balance=250),
-])
-session.commit()
-
-# Print out the balances.
-for account in session.query(Account):
-    print(account.id, account.balance)
+# The code below (1) checks the DB for existing IDs, and (2) generates
+# random IDs for new accounts, while ensuring the new IDs don't
+# collide with any existing ones.
+
+# Get and store existing account IDs for later collision checks during
+# random account ID creation.
+
+res = engine.execute('SELECT id from accounts;')
+seen = {}
+for elem in res:
+    val = elem[0]
+    if val not in seen:
+        seen[val] = 1
+
+
+def create_random_accounts(sess, seen, n):
+    """Create N new accounts with random IDs and random account balances.
+
+    Note that we do some work to avoid account ID collisions by
+    inspecting the contents of the SEEN dictionary.
+    """
+    new_accounts = []
+    elems = iter(range(n))
+    for i in elems:
+        maybe_id = floor(random.random()*1_000_000)
```

Comment:
This syntax is only supported in Python 3.6; we should call out the required version somewhere (or I think if we removed the underscores it could work in more versions).

Reply:
Fixed by removing the underscores and adding a note that Python 3 is used for the example code.